Mapping/DisconnectedMetadataFactory.php

```diff
@@ -126,8 +126,19 @@ public function findNamespaceAndPathForMetadata(ClassMetadataCollection $metadat
             throw new \RuntimeException(sprintf('Unable to determine where to save the "%s" class (use the --path option).', $all[0]->name));
         }
 
+        // When class exists use it's namespace
+        if (isset($r)) {
+            $ns = $r->getNamespaceName();
```

Comment:
this should be done inside the existing ``if (class_exists($all[0]->name)) {`` instead of checking whether the ``$r`` variable is set